### ISSUE TITLE
char8_t does not correctly show up in debugger on windows

### ISSUE THREAD

**Issue by darkhpp** [llvm/llvm-project#56184]
Clang on windows does not appear to be correctly generating debugging info for `char8_t`. In the VS debugger, it shows `???`. In the lldb(the one bundled with clion), the `_Ptr` and `_Buf` are outright missing. Both debuggers cannot show `char8_t[]` local variable as well. Code snippet is from Microsoft's STL's string. 

Works fine under `cl.exe`.

May be related: https://developercommunity.visualstudio.com/t/c-debugger-char8-t-deficiencies/739017

```cpp
template<typename T>
struct Str
{
    union _Bxty
    {
        T _Buf[16];
        T* _Ptr;
    } _Bx;
};

int main()
{
    char ra[] = "AAA";
    char8_t rb[] = u8"BBB";

    Str<char> a = { ._Bx = {._Ptr = &ra[0] } };
    Str<char8_t> b = { ._Bx = {._Ptr = &rb[0] } };
}
```

VS debugger:
![image](https://user-images.githubusercontent.com/26618520/175341059-7e7e7c12-e341-469f-bf8c-dcfdcda43828.png)

LLDB:
```
(lldb) p ra
(char [4]) $1 = "AAA"
(lldb) p rb
error: no member named 'rb' in namespace '$__lldb_local_vars'
error: use of undeclared identifier 'rb'
(lldb) p a
(Str<char>) $2 = {
  _Bx = (_Ptr = "AAA", _Buf = "\x94�/v\\")
}
(lldb) p b
(Str<char8_t>) $3 = (_Bx = Str<char8_t>::_Bxty @ 0x0000018dbfd95950)
```

Versions:
```
VS2022 17.2.4
LLVM 14.0.1
lldb version 9.0.0 (JetBrains IDE bundle; x64; build 152)
```

**Comment by EugeneZelenko** [comment]
It'll be good idea to use same version of Clang and LLDB.

**Comment by darkhpp** [comment]
Tested with LLVM 14.0.4:
```
(lldb) p ra
error: strs.exe :: Class 'std::strong_ordering' has a member 'less' of type 'const std::strong_ordering' which does not have a complete definition.
error: strs.exe :: Class 'std::partial_ordering' has a member 'less' of type 'const std::partial_ordering' which does not have a complete definition.
error: strs.exe :: Class 'std::weak_ordering' has a member 'less' of type 'const std::weak_ordering' which does not have a complete definition.
(char[4]) $0 = "AAA"
(lldb) p rb
error: strs.exe :: Class 'std::locale::_Locimp' has a base class 'std::locale::facet' which does not have a complete definition.
error: strs.exe :: Class '__crt_cached_ptd_host::cached<int>::guard' has a member '_copy' of type '__crt_cached_ptd_host::cached<int>' which does not have a complete definition.
error: strs.exe :: Class 'tagARRAYDESC' has a member 'tdescElem' of type 'tagTYPEDESC' which does not have a complete definition.
error: strs.exe :: Class 'tagPARAMDESCEX' has a member 'varDefaultValue' of type 'tagVARIANT' which does not have a complete definition.
error: expression failed to parse:
error: <lldb wrapper prefix>:45:31: no member named 'rb' in namespace '$__lldb_local_vars'
    using $__lldb_local_vars::rb;
          ~~~~~~~~~~~~~~~~~~~~^
error: <user expression 1>:1:1: use of undeclared identifier 'rb'
rb
^
(lldb) p a
(Str<char>) $1 = {
  _Bx = (_Buf = "\xd4\xfc\xaf\xba\xda\0\0\0 jey\xf7\U0000007f\0", _Ptr = "AAA")
}
(lldb) p b
(Str<char8_t>) $2 = (_Bx = Str<char8_t>::_Bxty @ 0x0000014b72e37360)
(lldb) version
lldb version 14.0.4
```

**Comment by labath** [comment]
I suspect this happens because the pdb reader does not recognise the char8_t in the pdb info. @ZequanWu might be interested in looking at that.

**Comment by ZequanWu** [comment]
Hi, I tested locally with thunk lldb, it works. I think this was fixed at  5c9e20d7d0a71439a95875ba6067f9c0fc7a4e04, which is not part of llvm 14.0.4.

**Comment by darkhpp** [comment]
Yep, that commit fixes it. Thanks for help.
